File: БР1.1/Цветкова Татьяна/labs/lab1/code/src/controllers/AuthController.ts
import { Request, Response } from "express";
import { AuthService } from "../services/AuthService";
import { LoginDto, RefreshDto, RegisterDto } from "../dto/auth.dto";
import { validateDto } from "../utils/validate";

const service = new AuthService();

export const AuthController = {
  async register(req: Request, res: Response) {
    const dto = await validateDto(RegisterDto, req.body);
    const result = await service.register(dto);
    res.status(201).json(result);
  },

  async login(req: Request, res: Response) {
    const dto = await validateDto(LoginDto, req.body);
    const result = await service.login(dto);
    res.json(result);
  },

  async refresh(req: Request, res: Response) {
    const dto = await validateDto(RefreshDto, req.body);
    const result = await service.refresh(dto.refreshToken);
    res.json(result);
  },
};

File: БР1.1/Захматов Юрий/labs/lab2/auth-service/src/main/java/com/petproject/authservice/controllers/AuthController.java
package com.petproject.authservice.controllers;


import com.petproject.authservice.dto.*;
import com.petproject.authservice.service.AuthService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping("/api/v1/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login")
    public ResponseEntity<AuthResponse> login(@RequestBody @Valid AuthRequest request) {
        AuthResponse response = authService.authenticate(request);

        return ResponseEntity.status(HttpStatus.OK).body(response);
    }

    @PostMapping("/register")
    public ResponseEntity<AuthResponse> register(@RequestBody @Valid RegisterRequest request) {
        AuthResponse response = authService.register(request);

        return ResponseEntity.status(HttpStatus.OK).body(response);
    }


    @PostMapping("/refresh")
    public ResponseEntity<RefreshTokenResponse> refresh(@RequestBody @Valid RefreshTokenRequest request) {
        RefreshTokenResponse response = authService.refresh(request);

        return ResponseEntity.status(HttpStatus.OK).body(response);
    }

    @PostMapping("/logout")
    public ResponseEntity<Void> logout(@RequestBody @Valid LogoutRequest request) {
        authService.logout(request);
        return ResponseEntity.status(HttpStatus.OK).body(null);
    }


}
